### PR TITLE
Adds share map functionality when using infowindow

### LINKED ISSUE #928: When infowindow is infowindow no sharelink object zoom.
Found one more bug, when infowindow is infowindow the zoom to object wont work at all. No id in url is present

_Originally posted by @mulfvik in https://github.com/origo-map/origo/pull/917#issuecomment-627834451_

### PR DESCRIPTION
**Pull request by jokd**
Fixes #928

### CODE REVIEW

File: src/permalink/permalinkstore.js [hunk 1]
```diff
@@ -45,6 +45,14 @@ permalinkStore.getState = function getState(viewer, isExtended) {
 
   if (featureinfo.getSelection().id && (type === 'AGS_FEATURE' || type === 'WFS' || type === 'GEOJSON' || type === 'TOPOJSON')) {
     state.feature = featureinfo.getSelection().id;
+  } else {
+    const selectedItems = viewer.getSelectionManager().getSelectedItems().getArray();
+    if (selectedItems.length > 0) {
+      const layerType = selectedItems[0].getLayer().getProperties().type;
+      if(layerType === 'AGS_FEATURE' || layerType === 'WFS' || layerType === 'GEOJSON' || layerType === 'TOPOJSON') {
+        state.feature = selectedItems[0].getId();
```
Comment (tonnyandersson) [inline src/permalink/permalinkstore.js:53]: `getId()` only returns id without layer name, at least when testing with default config. Feature part of sharemap url turns out like `&feature=8`.

**Comment by jokd** [comment]
You are right, I just tested our production config. A local geojson file obviously needs some other treatment.

**Comment by jokd** [comment]
Haven't tested with AGS or TOPOJSON services

**Comment by steff-o** [comment]
I don't think that this solution will work if using the _id_ parameter on WFS layers.

When creating the link the string is set to `&FEATURE=<layerName>.<PK>` and then when the link is restored it is assumed that `layerName` is in fact the the name of the layer on the map server when creating the id of the feature as `layername` is used both for identifying the layer and creating the id of the feature. If using the _id_ parameter on the layer this assumption will fail.

This is however probably not a new issue and consistent with current implementation. 

The simple solution would of course be to ignore the significance in WFS id and just use `<layerName>.<id>` for all layer types. That would unfortunately break old share links.

To solve it it would be necessary to check if the layer has an _id_ parameter and use that as first part of id instead of layerName when the link is restored.

**Comment by jokd** [comment]
I'd say we have always used the layername for this feature so I think it'll work with old links as well.

**Comment by steff-o** [comment]
Yes it will work with old links the way it always have worked. Just thought it would be a great opportunity to fix it for when using the _id_ parameter on WFS layers as that probably never have worked. Haven't tested though so I will drop it for now and maybe come back with a new issue.

**Comment by tonnyandersson** [comment]
The _zoom to_ animation is not what it used to be.

**Before**


**After**


I think this is caused by two `fit()` functions working against each other.

**Comment by jokd** [comment]
> The _zoom to_ animation is not what it used to be.
> 
> **Before** 
> 
> **After** 
> 
> I think this is caused by two `fit()` functions working against each other.

Yeah I am planning on making the one in showFeatureInfo optional when this and #1436 are merged. Then it will hopefully be a bit more smooth.

**Comment by tonnyandersson** [comment]
Ok, I'll allow it. For now.

**Review by tonnyandersson — APPROVED** [review]

**Pull request merged**